test/mcscupgrade/mcscupgrade_test.go

```diff
@@ -0,0 +1,403 @@
+
+			ocmMC, err := osdProvider.ClustersMgmt().V1().Clusters().Cluster(osdFleetManagerMC.Body().ClusterManagementReference().ClusterId()).Get().SendContext(ctx)
+			Expect(err).ShouldNot(HaveOccurred(), "failed to get get management cluster: %s", mcCluster.osdFleetMgmtID)
+
+			mcCluster.id = ocmMC.Body().ID()
+			mcCluster.name = ocmMC.Body().Name()
+			mcCluster.clusterVersion, err = semver.NewVersion(ocmMC.Body().Version().RawID())
+			Expect(err).ShouldNot(HaveOccurred(), "failed to parse management cluster installed version to semantic version")
+
+			availableVersions := ocmMC.Body().Version().AvailableUpgrades()
+			totalUpgradeVersionsAvailable := len(availableVersions)
+			Expect(totalUpgradeVersionsAvailable).ToNot(BeNumerically("==", 0), "management cluster has no available supported upgrade versions")
+
+			for i := 0; i < totalUpgradeVersionsAvailable; i++ {
+				version, err := semver.NewVersion(availableVersions[totalUpgradeVersionsAvailable-i-1])
+				Expect(err).ShouldNot(HaveOccurred(), "failed to parse management cluster upgrade version to semantic version")
+				if (mcCluster.clusterVersion.Minor() == version.Minor()) && upgradeType == "Z" {
+					mcCluster.upgradeVersion = version
+					break
+				} else if (mcCluster.clusterVersion.Minor() < version.Minor()) && upgradeType == "Y" {
+					mcCluster.upgradeVersion = version
+					break
+				}
+			}
+			Expect(mcCluster.upgradeVersion).ToNot(BeNil(), "failed to identify service cluster %q upgrade version", mcCluster.osdFleetMgmtID)
+
+			mcCluster.kubeconfigFile, err = osdProvider.KubeconfigFile(ctx, mcCluster.id)
+			Expect(err).ShouldNot(HaveOccurred(), "failed to get management cluster %q kubeconfig file", mcCluster.osdFleetMgmtID)
+
+			mcCluster.client, err = openshiftclient.NewFromKubeconfig(mcCluster.kubeconfigFile)
+			Expect(err).ShouldNot(HaveOccurred(), "failed to construct management cluster client")
+		}
+
+		if applyHCPWorkloads.MatchesLabelFilter(GinkgoLabelFilter()) {
+			hcpCluster.id, err = rosaProvider.CreateCluster(ctx, &rosaprovider.CreateClusterOptions{
+				ClusterName:  hcpCluster.name,
+				Version:      hcpCluster.version,
+				ChannelGroup: hcpCluster.channelGroup,
+				HostedCP:     true,
+				Properties:   fmt.Sprintf("provision_shard_id:%s", hcpCluster.provisionShardID),
+				WorkingDir:   hcpCluster.reportDir,
+			})
+			Expect(err).ShouldNot(HaveOccurred(), "failed to create hosted control plane cluster")
+
+			hcpCluster.kubeconfigFile, err = rosaProvider.KubeconfigFile(ctx, hcpCluster.id)
+			Expect(err).ShouldNot(HaveOccurred(), "failed to get hosted control plane cluster %q kubeconfig file", hcpCluster.id)
+
+			hcpCluster.client, err = openshiftclient.NewFromKubeconfig(hcpCluster.kubeconfigFile)
+			Expect(err).ShouldNot(HaveOccurred(), "failed to construct hosted control plane cluster client")
+		}
+	})
+
+	_ = AfterAll(func(ctx context.Context) {
+		defer func() {
+			_ = osdProvider.Client.Close()
+		}()
```

Comment:
where we create the client, you can do a `DeferCleanup(osdProvider.Client.Close)`

Reply:
Thank you for this suggestion! Will make the changes.